modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -701,8 +701,13 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
     val caseNamesAndIsUnit =
       caseNames.zip(alts.map(_.member == UnionMember.UnitCase))
 
+    val shapeDocs = hints
```

Comment:
**question:** wait, how does this change actually help? I don't see anything union-related in the diff.

Reply:
The `Foo.scala` union gets its `@param str` doc comment removed. It should be removed because the doc comment should be placed on `case class StrCase`
<img width="901" alt="image" src="https://user-images.githubusercontent.com/1857826/214982130-7174f558-9eec-4eae-b247-3c185bcf126e.png">

Reply:
(sorry for the screenshot, I don't know how to do a fancy diff block :P)

Reply:
I agree with the reasoning, but transformation is something that should happen in `SmithyToIR`, let's move it over there.

Reply:
I've been pondering over how I would do that, without it getting a bit funky.
I think that would require that `Documentation#shapeDocs` would be a `Hint[Documentation]` rather than a `List[String]` no?

Reply:
Indeed

Reply:
I actually can't figure out how to transform a `Option[DocumentationTrait]` into a `Hint.Documentation` where `shapeDocs` is a `Hint.Documentation` as well:
```
    shape.getTrait(classOf[DocumentationTrait]).asScala.map { doc =>
      val shapeDocs = Hint.Documentation(split(doc.getValue()), Map()) // Can't chuck in a List[String] in shapeDocs anymore
      Hint.Documentation(shapeDocs, memberDocs)
```

Reply:
Do I just introduce a new type, like `Hint.ShapeDocumentation`?

Reply:
I've given it some thoughts, and I think the most straightforward way is to keep the types as they are, but change [this line](https://github.com/disneystreaming/smithy4s/pull/731/files#diff-3d87f30f2b03db7c8e012d2c726b1e1a0f1227fe70d2e5c24166a9046c4428a5R758) to add a condition like `if(shape.isUnion) Map.empty else ...`, that'll be sufficient

Reply:
So simple, love it! :P

Almost makes me think that it's too simple to work.